options/options.c

```diff
@@ -150,6 +150,7 @@ static const m_option_t mp_vo_opt_list[] = {
     {"video-margin-ratio-right", OPT_FLOAT(margin_x[1]), M_RANGE(0.0, 1.0)},
     {"video-margin-ratio-top", OPT_FLOAT(margin_y[0]), M_RANGE(0.0, 1.0)},
     {"video-margin-ratio-bottom", OPT_FLOAT(margin_y[1]), M_RANGE(0.0, 1.0)},
+    {"video-crop", OPT_RECT(video_crop)},
```

Comment:
Mentioned this in IRC but I'll document it here so it doesn't get lost. All you need to do is add `.flags = UPDATE_IMGPAR` to this option and all the extra complicated runtime becomes unnecessary since mpv will automatically refresh the video.